### ISSUE TITLE
HTTPS EDNS Healthcheck not working, protocol mismatch.

### ISSUE THREAD

**Issue by michaelhajjar** [F5Networks/k8s-bigip-ctlr#2025]
#### Setup Details
 CIS Version : 2.5.0/2.5.1      
 Build: f5networks/k8s-bigip-ctlr:latest       
 BIGIP Version: BIG-IP 16.0.1.1 Build 0.0.6 Point Release 1
 AS3 Version: 3.26.1-1      
 Agent Mode: AS3/CCCL       
 Orchestration: OpenShift  
 Orchestration Version:  Kubernetes version: v1.20.0+558d959, OpenShift version: 4.7.21
 Pool Mode: Nodeport      
 Additional Setup details: -

#### Description  
The EDNS Healtcheck seems to fail when deploying a HTTPS application.
It fails with the error: Protocol mismatch (see screenshot below).
![image](https://user-images.githubusercontent.com/37404682/135993153-4c119d22-ecc5-4da7-ad21-946190ce7a28.png)
 
#### Steps To Reproduce
1) Extract the tar.gz helm chart.
2) run: helm template . > deployment.yaml.
3) run: kubectl/oc create -f deployment.yaml.
4) Set the last resort pool manually to some pool because of bug (https://github.com/F5Networks/k8s-bigip-ctlr/issues/1792).
4) Login to the LTM/GSLB, it should have new records. Check the GSLB WideIP -> Pool -> Member. Check the health status, it should say protocol mismatch.

#### Expected Result 
A healthy EDNS/Wide record HTTPS application with no faulty checks. 

#### Actual Result  
Not healthy EDNS/Wide record HTTPS application.

#### Diagnostic Information
```buildoutcfg
No error messages in the logging of the CIS-controller that seems to stand out. 
Note: Sanitize the data. For example, be mindful of IPs, ports, application names and URLs
Note: The following F5 article outlines the information required when opening an issue.
https://support.f5.com/csp/article/K60974137
```  

#### Observations (if any)
HTTP EDNS healthchecks are working correctly. 

#### Attachments
[example-application-helm.tar.gz](https://github.com/F5Networks/k8s-bigip-ctlr/files/7284567/example-application-helm.tar.gz)

**Comment by gwolfis** [comment]
I just finished testing the customer setup and was able to repodruce the error. HTTPS monitor shows 'state: protocol mismatch'. HTTP setup works as expected.

**Comment by mdditt2000** [comment]
Filed Jira issue CONTCNTR-2938 for PME tracking. ETA is CIS 2.6.1

**Comment by trinaths** [comment]
Created JIRA CONTCNTR-3056 for internal tracking.

**Comment by mdditt2000** [comment]
SNI is missing 

![image](https://user-images.githubusercontent.com/12733938/145620378-c617c6a6-9ffc-469c-ba0c-78eb2264f580.png)

**Comment by mdditt2000** [comment]
Issue is resolved with adding the correct SNI. BIG-IP version 16.1.x is required. @gwolfis @michaelhajjar would you like a image to test? Fix will be CIS 2.7

**Comment by gwolfis** [comment]
Hi Mark,

Yes please.

Gert

Excuse my errors, this message was sent from a mobile device.

________________________________
Van: mark dittmer ***@***.***>
Verzonden: Monday, December 13, 2021 7:45:36 PM
Aan: F5Networks/k8s-bigip-ctlr ***@***.***>
CC: Gert Wolfis ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [F5Networks/k8s-bigip-ctlr] HTTPS EDNS Healthcheck not working, protocol mismatch. (#2025)


EXTERNAL MAIL: ***@***.***


Issue is resolved with adding the correct SNI. BIG-IP version 16.1.x is required. @gwolfis<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgwolfis&data=04%7C01%7CG.Wolfis%40F5.com%7C46c83e67428448e22d9808d9be68d9a2%7Cdd3dfd2f6a3b40d19be0bf8327d81c50%7C0%7C0%7C637750179820304273%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=WVNrrCtjHLgXpJ9CdJAru3qE3rpsJVF4BVWZZjkG5XA%3D&reserved=0> @michaelhajjar<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmichaelhajjar&data=04%7C01%7CG.Wolfis%40F5.com%7C46c83e67428448e22d9808d9be68d9a2%7Cdd3dfd2f6a3b40d19be0bf8327d81c50%7C0%7C0%7C637750179820304273%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=LKyl%2FeWwhFxLXQ%2F379XSY4jAZEWp3%2FM3oljaVe1QGBQ%3D&reserved=0> would you like a image to test? Fix will be CIS 2.7

**Comment by mdditt2000** [comment]
@gwolfis https://github.com/mdditt2000/kubernetes-1-19/blob/master/cis%202.7/edns/cis-deployment/f5-cluster-deployment.yaml
